### PR TITLE
wip thematize specialty icons

### PR DESCRIPTION
**Pull request by akasolace**
1. changes proposed in this pull request:
 
   - fixes issue #766
   
2. changelog and release_notes ...

 - [ ] have been updated
 - [x] do not require update

### CODE REVIEW

File: src/main/java/core/gui/theme/ImageUtilities.java [hunk 1]
```diff
@@ -581,6 +581,19 @@ public static Icon getSmileyIcon(String smileyName) {
 		return null;
 	}
 
+	public static Icon getSmallPlayerSpecialtyIcon(String playerSpecialtyName) {
+		return getPlayerSpecialtyIcon(playerSpecialtyName, 15);
+	}
+
+	public static Icon getPlayerSpecialtyIcon(String playerSpecialtyName, int size) {
+		if (Arrays.stream(HOIconName.SPECIALTIES).skip(1).anyMatch(playerSpecialtyName::equals)) {
```
Comment (tychobrailleur) [inline src/main/java/core/gui/theme/ImageUtilities.java:589]: I believe `ThemeManager` was caching those images, we probably should make sure those ones are cached as well, as it has caused display perf issues in the past.

Comment (akasolace) [reply src/main/java/core/gui/theme/ImageUtilities.java:594]: I have another issue at the moment with this one that I can't explain in Lineup tab.

**Before:**

image


With this commie:
image


The bottom line is truncated. I can't figure out what I did that is causing it. Looks like the preferred size is now wrongly calculated.

Comment (akasolace) [reply src/main/java/core/gui/theme/ImageUtilities.java:594]: if you have an idea of what is going on ....

Comment (tychobrailleur) [reply src/main/java/core/gui/theme/ImageUtilities.java:594]: I have the same problem after switching to darklaf 2.5.x -- reverting back to 2.4.x fixes the issue.

Comment (akasolace) [reply src/main/java/core/gui/theme/ImageUtilities.java:594]: >I have the same problem after switching to darklaf 2.5.x -- reverting back to 2.4.x fixes the issue

indeed you are right ...  @weisJ any idea what is going on?

Comment (akasolace) [reply src/main/java/core/gui/theme/ImageUtilities.java:600]: > I believe ThemeManager was caching those images, we probably should make sure those ones are cached as well, as it has caused display perf issues in the past.

good catch ... I added the caching

**Comment by akasolace** [comment]
@tychobrailleur I think it is ready for merge. In parallel I created #771 for the rendering of playerCB

**Pull request merged**